NitroxClient/MonoBehaviours/NitroxEntity.cs

```diff
@@ -76,16 +77,77 @@ public static bool TryGetObjectFrom(NitroxId id, out GameObject gameObject)
         public static bool TryGetEntityFrom(GameObject gameObject, out NitroxEntity nitroxEntity)
         {
             nitroxEntity = null;
-            return gameObject != null && gameObject.TryGetComponent(out nitroxEntity);
+            return gameObject && gameObject.TryGetComponent(out nitroxEntity);
         }
 
+        public static bool TryGetIdFrom(GameObject gameObject, out NitroxId nitroxId)
+        {
+            if (gameObject && gameObject.TryGetComponent(out NitroxEntity nitroxEntity))
+            {
+                nitroxId = nitroxEntity.Id;
+                return true;
+            }
+
+            nitroxId = null;
+            return false;
+        }
+
+        public static bool TryGetIdFrom(Component component, out NitroxId nitroxId)
+        {
+            if (component && component.gameObject.TryGetComponent(out NitroxEntity nitroxEntity))
+            {
+                nitroxId = nitroxEntity.Id;
+                return true;
+            }
+
+            nitroxId = null;
+            return false;
+        }
+
+        public static bool TryGetIdOrWarn<T>(GameObject gameObject, out NitroxId nitroxId) where T : class
+        {
+            if (gameObject && gameObject.TryGetComponent(out NitroxEntity nitroxEntity))
+            {
+                nitroxId = nitroxEntity.Id;
+                return true;
+            }
+
+            Log.Warn($"[{typeof(T).Name}] Couldn't find an id on {gameObject.GetFullHierarchyPath()}");
+            nitroxId = null;
+            return false;
+        }
```

Comment:
Having to pass in a `T` for the calling class makes it quite verbose. I recommend using `CallerFilePath` as a last parameter instead.

Reply:
That `CallerFilePath` is really nice.
If I have to create NitroxEntity from the ground up I would move the id getting to GameObjectExtension. But I wouldn't do so now because all the id logic is located in NitrroxEntity. Using a static method also highlights the method because NitroxEntity gets colored a different color in the IDE.